pkg/resources/table_acceptance_test.go

```diff
@@ -2090,3 +2092,185 @@ resource "snowflake_table" "test_table" {
+		},
+		Steps: []resource.TestStep{
+			{
+				ProtoV6ProviderFactories: acc.TestAccProtoV6ProviderFactories,
+				Config:                   tableConfigIssue3007(name, databaseName, schemaName, "NUMBER"),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(resourceName, "column.0.type", "NUMBER(11,2)"),
+					resource.TestCheckResourceAttr(resourceName, "column.1.type", "NUMBER(38,0)"),
+				),
+			},
+			{
+				ProtoV6ProviderFactories: acc.TestAccProtoV6ProviderFactories,
+				Config:                   tableConfigIssue3007(name, databaseName, schemaName, "NUMBER(11)"),
+				ConfigPlanChecks: resource.ConfigPlanChecks{
+					PreApply: []plancheck.PlanCheck{
+						planchecks.ExpectChange(resourceName, "column.1.type", tfjson.ActionUpdate, sdk.String("NUMBER(38,0)"), sdk.String("NUMBER(11)")),
+					},
+				},
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(resourceName, "column.0.type", "NUMBER(11,2)"),
+					resource.TestCheckResourceAttr(resourceName, "column.1.type", "NUMBER(11,0)"),
+				),
+			},
+			{
+				ProtoV6ProviderFactories: acc.TestAccProtoV6ProviderFactories,
+				Config:                   tableConfigIssue3007(name, databaseName, schemaName, "NUMBER"),
+				ConfigPlanChecks: resource.ConfigPlanChecks{
+					PreApply: []plancheck.PlanCheck{
+						planchecks.ExpectChange(resourceName, "column.1.type", tfjson.ActionUpdate, sdk.String("NUMBER(11,0)"), sdk.String("NUMBER")),
+					},
+				},
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(resourceName, "column.1.type", defaultNumber),
+				),
+			},
+			{
+				ProtoV6ProviderFactories: acc.TestAccProtoV6ProviderFactories,
+				Config:                   tableConfigIssue3007(name, databaseName, schemaName, defaultNumber),
+				ConfigPlanChecks: resource.ConfigPlanChecks{
+					PreApply: []plancheck.PlanCheck{
+						plancheck.ExpectResourceAction(resourceName, plancheck.ResourceActionNoop),
+					},
+				},
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(resourceName, "column.1.type", defaultNumber),
+				),
+			},
+			{
+				ProtoV6ProviderFactories: acc.TestAccProtoV6ProviderFactories,
+				Config:                   tableConfigIssue3007(name, databaseName, schemaName, "decimal"),
+				ConfigPlanChecks: resource.ConfigPlanChecks{
+					PreApply: []plancheck.PlanCheck{
+						plancheck.ExpectResourceAction(resourceName, plancheck.ResourceActionNoop),
+					},
+				},
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(resourceName, "column.1.type", defaultNumber),
+				),
+			},
+		},
+	})
+}
+
+func tableConfigIssue3007(name string, databaseName string, schemaName string, dataType string) string {
+	return fmt.Sprintf(`
+resource "snowflake_database" "test_database" {
+    name = "%[2]s"
+}
+
+resource "snowflake_schema" "test_schema" {
+    depends_on = [snowflake_database.test_database]
+    name = "%[3]s"
+    database = "%[2]s"
+}
```

Comment:
Why not default schema?

Reply:
It was a copy-pasted config; adjusted and used default schema and the db